packages/react-app/src/components/BuilderBadgeCard.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useEffect } from "react";
+import { Flex, Skeleton, SkeletonText } from "@chakra-ui/react";
+import useCustomColorModes from "../hooks/useCustomColorModes";
+import axios from "axios";
+
+const BuilderBadgeCardSkeleton = ({ isLoaded, children }) => (
+  <Skeleton isLoaded={isLoaded}>{isLoaded ? children() : <SkeletonText mt="4" noOfLines={4} spacing="4" />}</Skeleton>
+);
+
+// TODO get the actual join date. Should be easy getting the user.create event
+const BuilderBadgeCard = ({ builder, readContracts, builderAddress }) => {
+    const [balance, setBalance] = useState({
+      loading: true,
+      items: [],
+    });
+
+    const { borderColor, secondaryFontColor } = useCustomColorModes();
+
+    const getImageAddress = async ( badge ) => {
+      const metadata = await axios.get(`https://forgottenbots.mypinata.cloud/ipfs/QmZesNT9tbpaNoy727fYRa7cB936dznKqFtZwNwUSbxJBg/000000000000000000000000000000000000000000000000000000000000000${badge}.json`);
+      return metadata.data
+    }
+  
+  const loadBadges = async () => {
+    if (!builderAddress || !readContracts) return;
+    setBalance({
+      loading: true,
+      items: [],
+    });
+
+    var balances = [];
+
+    for (let i = 0; i < 10; i += 1) {
+      let userBalance = (await readContracts.BuidlBadges.balanceOf(builderAddress, i)).toNumber();
+      //console.log(userBalance)
+      if (userBalance > 0) {
+      balances.push(i)
+      console.log(balances)
+      }
+    }
+
```

Comment:
Two things here:

1. A question: it would be possible to avoid iterating though all the badges? Maybe having on the contract a mapping with an array (address => array of numbers)? (I'm not fluent in Solidity :D). That way it'll only require a contract call per user.

2. Converting a BigNumber to a JS number (toNumber()) can lead to overflow errors, so we should use the Big Number built-in operators (gt, lt, etc). Fixed on https://github.com/moonshotcollective/scaffold-directory/pull/120/commits/391d51a80f629a04b5a9a62c100db0a321b862c1

Reply:
1. Mappings in solidity aren’t iterable, but there is a solution. I’ll update.
2. TY was not aware